src/main/java/com/yahoo/identity/services/storage/sql/AccountModel.java

```diff
@@ -1,52 +1,141 @@
 package com.yahoo.identity.services.storage.sql;
 
+import static com.kosprov.jargon2.api.Jargon2.Hasher;
+import static com.kosprov.jargon2.api.Jargon2.jargon2Hasher;
+import static com.kosprov.jargon2.api.Jargon2.jargon2Verifier;
+
+import com.kosprov.jargon2.api.Jargon2;
+
+import java.security.SecureRandom;
+import java.time.Instant;
+
+import javax.annotation.Nonnull;
 
 public class AccountModel {
+
     private String uid;
     private String username;
     private String firstName;
     private String lastName;
     private String email;
-    private Boolean verified;
-    private String password;
+    private int emailStatus;
+    private String passwordHash;
+    private String passwordSalt;
     private String description;
     private long createTs;
     private long updateTs;
+    private long blockUntilTs;
+    private int consecutiveFails;
 
-    public String getUid() { return this.uid; }
+    public String getUid() {
+        return this.uid;
+    }
 
-    public String getUsername() { return this.username; }
+    @Nonnull
+    public String getUsername() {
+        return this.username;
+    }
 
-    public void setUsername(String username) { this.username = username; }
+    public void setUsername(@Nonnull String username) {
+        this.username = username;
+    }
 
-    public String getFirstName() { return this.firstName; }
+    public String getFirstName() {
+        return this.firstName;
+    }
 
-    public void setFirstName(String firstName) { this.firstName = firstName; }
+    public void setFirstName(@Nonnull String firstName) {
+        this.firstName = firstName;
+    }
 
-    public String getLastName() { return this.lastName; }
+    public String getLastName() {
+        return this.lastName;
+    }
 
-    public void setLastName(String lastName) { this.lastName = lastName; }
+    public void setLastName(@Nonnull String lastName) {
+        this.lastName = lastName;
+    }
 
-    public String getEmail() { return this.email; }
+    public String getEmail() {
+        return this.email;
+    }
 
-    public void setEmail(String email, Boolean verified) {
+    public void setEmail(@Nonnull String email) {
         this.email = email;
-        this.verified = verified;
     }
 
-    public String getDescription() { return description; }
+    public int getEmailStatus() {
+        return this.emailStatus;
+    }
+
+    public void setEmailStatus(@Nonnull int emailStatus) {
+        this.emailStatus = emailStatus;
+    }
+
+    public String getDescription() {
+        return this.description;
+    }
+
+    public void setDescription(@Nonnull String description) {
+        this.description = description;
+    }
+
+    public String getPasswordHash() {
+        return this.passwordHash;
+    }
+
+    public String getPasswordSalt() {
+        return this.passwordSalt;
+    }
+
+    public void setPassword(@Nonnull String password) {
+        SecureRandom secureRandom = new SecureRandom();
```

Comment:
`SecureRandom` may takes long time when initialized because it needs to wait for enough entrophy.
You can try to have only one `SecureRandom`, and reuse it. (some people say using one instance per thread (using ThreadLocal), you may also do that if you want, but not required)
May write a `RandomService` to take care of that.
This can be done in future PR.

Reply:
Thanks for the insight! Noted and I will implment `RandomService` in the next PR.